### PR TITLE
Add ability to pack variable numbers of tables into an MBE using RegEx structure names

### PR DESCRIPTION
**Pull request by Pherakki**

### CODE REVIEW

File: DSCSTools/EXPA.cpp [hunk 1]
```diff
@@ -184,16 +184,26 @@ namespace dscstools {
 			}
 
 			boost::property_tree::ptree format = getStructureFile(source);
+			auto fname = source.filename().string();
 
-			if (format.size() != tables.size())
-				throw std::runtime_error("Warning: number of tables doesn't match expected values. " + source.filename().string());
+			//if (format.size() != tables.size())
+			//	throw std::runtime_error("Warning: number of tables doesn't match expected values. " + fname);
```
Comment (SydMontague) [inline DSCSTools/EXPA.cpp:190]: Remove if not needed anymore

File: DSCSTools/EXPA.cpp [hunk 2]
```diff
@@ -264,11 +274,11 @@ namespace dscstools {
 
 			boost::property_tree::ptree format = getStructureFile(source);
 
-			std::size_t numTables = format.size();
+			std::size_t numTables = 0;// format.size();
 
 			// write EXPA Header
 			output.write("EXPA", 4);
-			output.write(reinterpret_cast<char*>(&numTables), 4);
+			output.write(reinterpret_cast<char*>(&numTables), 4); // Going to overwrite this later
```
Comment (SydMontague) [inline DSCSTools/EXPA.cpp:281]: remove it if you're writing it later

File: DSCSTools/EXPA.cpp [hunk 3]
```diff
@@ -264,11 +274,11 @@ namespace dscstools {
 
 			boost::property_tree::ptree format = getStructureFile(source);
 
-			std::size_t numTables = format.size();
+			std::size_t numTables = 0;// format.size();
```
Comment (SydMontague) [inline DSCSTools/EXPA.cpp:277]: Remove commented code

File: DSCSTools/EXPA.cpp (hunk 2 above)
Comment (Pherakki) [reply DSCSTools/EXPA.cpp:281]: Would you prefer to seek ahead by 4 bytes in this line or explicitly write 0 (current behaviour)?

**Pull request merged**